boa_engine/src/syntax/ast/expression/call.rs

```diff
@@ -0,0 +1,136 @@
+use crate::syntax::ast::{join_nodes, ContainsSymbol};
+use boa_interner::{Interner, ToInternedString};
+
+use super::Expression;
+
+/// Calling the function actually performs the specified actions with the indicated parameters.
+///
+/// Defining a function does not execute it. Defining it simply names the function and
+/// specifies what to do when the function is called. Functions must be in scope when they are
+/// called, but the function declaration can be hoisted. The scope of a function is the
+/// function in which it is declared (or the entire program, if it is declared at the top
+/// level).
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://tc39.es/ecma262/#prod-CallExpression
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions#Calling_functions
+#[cfg_attr(feature = "deser", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Debug, PartialEq)]
+pub struct Call {
+    target: Box<Expression>,
+    args: Box<[Expression]>,
+}
+
+impl Call {
+    /// Creates a new `Call` AST Expression.
+    pub fn new(target: Expression, args: Box<[Expression]>) -> Self {
+        Self {
+            target: target.into(),
+            args,
+        }
+    }
+
+    /// Gets the name of the function call.
+    pub fn expr(&self) -> &Expression {
+        &self.target
+    }
+
+    /// Retrieves the arguments passed to the function.
+    pub fn args(&self) -> &[Expression] {
+        &self.args
+    }
```

Comment:
We seem to be adding `#[inline]` on the other methods, and these are getters/constructors, so it makes sense to inline them, right?

Same happens in other nodes.

Reply:
Let me see if I can make Clippy do this for me